src/Cardano/Wallet/CoinSelection/Fee.hs

```diff
@@ -341,3 +350,62 @@ computeFee (CoinSelection inps outs chgs) =
     -- by definition, impossible; unless we messed up real hard.
     collapse [] _ =
         invariant "outputs are bigger than inputs" (undefined) (const False)
+
+-- | A linear equation on the transaction size. Represents the @\s -> a + b*s@
+-- function where @s@ is the transaction size in bytes, @a@ and @b@ are
+-- constant coefficients.
+data TxSizeLinear = TxSizeLinear Double Double
+    deriving (Eq, Show)
+
+cardanoPolicy :: TxSizeLinear
+cardanoPolicy = TxSizeLinear 155381 43.946
+
+-- | Estimate the fee for a transaction that has @ins@ inputs
+--   and @length outs@ outputs. The @outs@ lists holds the coin value
+--   of each output.
+--
+--   NOTE: The average size of @Attributes AddrAttributes@ and
+--         the transaction attributes @Attributes ()@ are both hard-coded
+estimateCardanoFee
+    :: TxSizeLinear
+    -> (Tx, [TxWitness])
+    -> Fee
+estimateCardanoFee (TxSizeLinear a b) txWithWitness@(Tx inps _, _) =
+    trace ("totalPayload: "<> show totalPayload
+          <> " length : " <> show (BL.length $ toLazyByteString $ encodeSignedTx txWithWitness)) $
+    Fee $ ceiling $ a + b*totalPayload
+    where
+        totalPayload :: Double
+        totalPayload = fromIntegral
+            $ boundAddrAttrSize + boundTxAttrSize + boundSignatureWitnessSize + payloadFromTxWithWitness + 4*(length inps - 1)
```

Comment:
where does this `4*` comes from ? Also, we should probably multiply the witness size by the number of witness (i.e. inputs) ...